src/db/migrate.rs

```diff
@@ -644,6 +644,94 @@ pub fn migrate(version: Option<Version>, conn: &mut Client) -> CratesfyiResult<(
             ALTER TABLE releases ADD COLUMN authors JSON;
             ",
         ),
+        migration!(
+            context,
+            28,
+            // description
+            "Add gitlab handling: creation of the new repositories table which replaces and extend \
+             github_repos",
+            // upgrade query
+            "
+                CREATE TABLE repositories (
+                    id SERIAL PRIMARY KEY,
+                    host VARCHAR NOT NULL,
+                    host_id VARCHAR NOT NULL,
+                    name VARCHAR NOT NULL,
+                    description VARCHAR,
+                    last_commit TIMESTAMPTZ,
+                    stars INT NOT NULL,
+                    forks INT NOT NULL,
+                    issues INT NOT NULL,
+                    updated_at TIMESTAMPTZ NOT NULL,
+                    UNIQUE (host, host_id)
+                );
+
+                ALTER TABLE releases ADD COLUMN repository_id INTEGER
+                    REFERENCES repositories(id) ON DELETE SET NULL;
+
+                INSERT INTO repositories(host, host_id, name, description, last_commit, stars, forks, issues, updated_at)
+                    SELECT 'github.com', id, name, description, last_commit, stars, forks, issues, updated_at
+                    FROM github_repos;
+
+                UPDATE releases
+                    SET repository_id = repositories.id
+                FROM repositories
+                WHERE releases.github_repo IS NOT NULL AND repositories.host_id = releases.github_repo;
+
+                DROP INDEX releases_github_repo_idx;
+                DROP INDEX github_repos_stars_idx;
+
+                CREATE INDEX releases_github_repo_idx ON releases(repository_id);
+                CREATE INDEX github_repos_stars_idx ON repositories(stars DESC);
```

Comment:
Why is "github" in the name here? Shouldn't it just be "repo"?
```suggestion
                CREATE INDEX releases_repo_idx ON releases(repository_id);
                CREATE INDEX repos_stars_idx ON repositories(stars DESC);
```

Reply:
Good point!

Reply:
I also changed it in the downgrade query.